### PR TITLE
integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by quavo19**
### In this Milestone, my partner and I;
Solved the N+1 problem when fetching all posts and their comments for a user.

Used Capybara to write integration tests for each view in your project which includes ;

 User index page

 User show page

 Post index page

 Post-show page

### CODE REVIEW

File: spec/integrations/posts_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    end
+
+    it 'renders likes count of a post' do
+      expect(page).to have_content(@fourth_post.likes_counter)
+      expect(page).to have_content(@second_post.likes_counter)
+      expect(page).to have_content(@third_post.likes_counter)
+    end
+
+    it "redirects to a specific post's show page" do
+      click_link @second_post.title
```
Comment (alexander16108) [inline spec/integrations/posts_index_spec.rb:67]: currently there is not test in your post index to 
`Tests that asserts you can see a section for pagination if there are more posts than fit on the view`

File: spec/integrations/users_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      expect(page).to have_content(@first_post.title)
+      expect(page).to have_content(@second_post.title)
+      expect(page).to have_content(@third_post.title)
+    end
+
+    it "redirects to user's posts page" do
+      click_link @first_post.title
+      expect(page).to have_current_path(user_post_path(@user, @first_post))
+    end
+  end
```
Comment (alexander16108) [inline spec/integrations/users_show_spec.rb]: currently there is not test in your user show to

``` Tests that asserts you can see a button that lets me view all of a user's posts.```
``` ```

**Review by alexander16108 — CHANGES_REQUESTED** [review]
Hi @quavo19 @techEmprez , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- screenshots ❌ 
- passing linters ✅ 
- good readme file ✅ 
- good pr ✅ 

## Required Changes ♻️
_Check the comments under the review._

- your PR description should include screenshots of your app's logs before and after the fix
- your readme file should have social media links to both authors of the project

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/integrations/users_show_spec.rb (hunk 2 above)
Comment (quavo19) [reply spec/integrations/users_show_spec.rb]: @alexander16108 you mean a tests to see just the button right ?

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @quavo19 , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- [ ] All linters passed

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**